ruby/ql/test/query-tests/security/cwe-601/UrlRedirect.rb

```diff
@@ -83,6 +83,11 @@ def route14
     redirect_back_or_to params[:key], allow_other_host: false
   end
 
+  # GOOD
+  def route15
+    redirect_to cookies[:foo]
```

Comment:
I'm not entirely sure cookies should be considered safe. It is indeed harder for an attacker to influence, however, https://cwe.mitre.org/data/definitions/601.html states:
> Many open redirect problems occur because the programmer assumed that certain inputs could not be modified, such as **cookies** and hidden form fields.
--
 

<br class="Apple-interchange-newline">

Reply:
I don't know, perhaps some funny `javascript:` URLs, but mitre mentions it in the CWE for some reason. Best ask the security lab folk.

Reply:
Sounds good to me.